### DISCUSSION TITLE
Option for larger size?

### DISCUSSION THREAD

**Discussion by peyre** [raceintospace/raceintospace#731] [Ideas]
The config file offers an option to full-size the game, which is helpful to those with giant monitors.  (I don't notice it myself since I use 17" monitors, but it's a problem for some others.)  I wonder if we could also add an option to make it double or triple size.  Maybe something like:

```
# "screensize", &options.set_size, "%u", 0,
# Set to 2 for double size, 3 for triple, etc.
# size 1
```

**Comment by peyre** [comment]
Or maybe the **fullscreen** option could be expanded:

```
# "fullscreen", &options.want_fullscreen, "%u", 0,
# "Set to 1 if you want (ugly) full-screen game."
# "Set to 2 or above if you want the game double size, triple size, etc.
```

(Though I see it's currently a boolean value, not an integer.)

**Comment by alsed** [comment]
The game has a native resolution of 320x200 and scales to 2x.
There is a constant attribute called SCALE (in the Display::Graphics class) set to 2, but the reality is that the upscaling was hardcoded.
I have removed the fixed values for native resolution and upscaling (pull request #849 & #850) , so now it is possible to upscale the resolution reasigning the value of SCALE.
But there is a tradeoff, at larger scale black lines start to appear between pixels, similar to scanlines. 

![Captura desde 2024-06-04 09-27-27](https://github.com/raceintospace/raceintospace/assets/7539842/add8caa4-3623-4d17-b77e-42f512aa9cff)
Scaled to 4x resolution.

I'm not completely sure, but it seems that there is no way to have assets at different resolutions in the game. So if we want to improve the resolutions of any image, we would have to do it for all the assets. A good task, but a huge one.

**Reply by peyre** [reply]
This looks like good work, Alsed.  I've merged your PRs.  Feel free to keep at this; it would be nice to be able to scale the game well - I might find myself running it at 2X like I do Sid Meier's Colonization (but there, DOSBox is doing the scaling).

How do you go _about_ scaling the game, btw?  Did you make it a setting in the config file?

**Reply by alsed** [reply]
Thank you Peyre, place a setting in the config file would be the best option, i would look into it.
If you want to try it, you can change the variable in src/display/graphics.h, change 'SCALE' to 3 and then compile.

**Reply by peyre** [reply]
Thanks, I just tried it out.  It worked, but I had to recompile the game to make the change work.  The config file would definitely be the place for it, since we have other advanced options there.

**Comment by alsed** [comment]
Hi @peyre 
Changes have been pull to the master to allow 4x screen scaling option. You can change it to 2x from the config file. 
If you can, please download my master fork and compile the source from there to test the changes.
https://github.com/Alsed/raceintospace/tree/master
